lcls-twincat-motion/Library/DUTs/DUT_MotionPneumaticActuator.TcDUT

```diff
@@ -0,0 +1,157 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4024.3">
+  <DUT Name="DUT_MotionPneumaticActuator" Id="{ee78164b-027c-4a52-afc0-269c58a42ceb}">
+    <Declaration><![CDATA[{attribute 'qualified_only'}
+{attribute 'strict'}
+TYPE DUT_MotionPneumaticActuator :
+    // Defines the EPICS interface to actuating a pneumatic stage
+STRUCT
+     (* Hardware *)
+    //Readbacks
+    //Limit Switch
+    {attribute 'pytmc' := '
+     pv: PLC:bInLimitSwitch
+     io: i
+     field: ZNAM FALSE
+     field: ONAM TRUE
+     field: DESC TRUE if IN limit is reached
+    '}
+    i_bInLimitSwitch    :    BOOL;
+    {attribute 'pytmc' := '
+     pv: PLC:bOutLimitSwitch
+     io: i
+     field: ZNAM FALSE
+     field: ONAM TRUE
+     field: DESC TRUE if OUT limit is reached
+    '}
+    i_bOutLimitSwitch    :    BOOL;
+    //Controls
+    //Digital outputs
+    {attribute 'pytmc' := '
+    pv: bRetractDigitalOutput;
+    io: i;
+    field: ONAM FALSE
+    field: ZNAM TRUE
+    field: DESC TRUE if Retract digital output is active
+    '}
+    q_bRetract    :    BOOL;
+    {attribute 'pytmc' := '
+    pv: bInsetDigitalOutput;
+    io: i;
+    field: ONAM FALSE
+    field: ZNAM TRUE
+    field: DESC TRUE if Insert digital output is active
+    '}
+    q_bInsert    :    BOOL;
+
+
+    //Logic and supervisory
+    {attribute 'pytmc' := '
+    pv: bInterlockOK;
+    io: i;
+    field: ZNAM FALSE
+    field: ONAM TRUE
+    field: DESC True if the actuator has permission to move in either direction
+    '}
+    bILK_OK: BOOL;
+    {attribute 'pytmc' := '
+    pv: bInsertEnable;
+    io: i;
+    field: ZNAM FALSE
+    field: ONAM TRUE
+    field: DESC True if the actuator had permission to be retracted
+    '}
+    bInsertOK    :    BOOL;
+    {attribute 'pytmc' := '
+    pv: bRetractEnable;
+    io: i;
+    field: ZNAM FALSE
+    field: ONAM TRUE
+    field: DESC True if the actuator had permission to be inserted
+    '}
+    bRetractOK    :    BOOL;
+
+    (* Commands *)
+    // Used from Epics to comand the actuator to  move
+    {attribute 'pytmc' := '
+    pv: IN_CMD;
+    io: io;
+    DESC Used by EPICS and internally to request Insert motion
+    '}
+    bInsert_SW        :    BOOL;
+    {attribute 'pytmc' := '
+    pv: OUT_CMD;
+    io: io;
+    DESC Used by EPICS and internally to request retract motion
```

Comment:
The two pragma fixes here got reverted somehow? Needs to be `field: DESC ...`

Reply:
Will fix it now.

And I don't really like the IN_CMD and OUT_CMD. Any suggestions?

Reply:
I was having a similar naming issue somewhere else when doing the states thing... I never found a great name for these command PVs. I ended up using `SET` for mine, but that doesn't work as well when you have two different commands... I'm not sure what would be best here. As implemented already it is at least clear what they do.

Reply:
Some other ideas (not necessarily better):
`OUT:SET`
`SET:OUT`
`GO:OUT`
`OUT:GO`
`OUT:CMD`
`CMD:OUT`

To at least get the underscore out of the picture

Reply:
CMD:OUT, CMD:IN?